### PR TITLE
Colorize messages when a player wants to do something behind barrier

### PR DESCRIPTION
**Pull request by DiamondPlane**
106389910-90583c00-63ee-11eb-9486-7252c0f4e499

### CODE REVIEW

File: mods/ctf/ctf_map/ctf_map_core/barrier.lua [hunk 1]
```diff
@@ -222,7 +222,8 @@ if minetest.get_modpath("ctf") then
 		local tname = ctf.player(name).team
 		if tname and
 				(tname == "blue" and pos.z >= 0) or (tname == "red" and pos.z <= 0) then
-			minetest.chat_send_player(name, "Can't dig beyond the barrier!")
+			minetest.chat_send_player(name,
+minetest.colorize("#e32d2d", "Can't dig beyond the barrier!"))
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_map/ctf_map_core/barrier.lua:226]: This should be indented two tabs past the start of line 225

**Review by LoneWolfHT — CHANGES_REQUESTED** [review]

File: mods/ctf/ctf_map/ctf_map_core/barrier.lua (hunk 1 above)
Comment (LoneWolfHT) [reply mods/ctf/ctf_map/ctf_map_core/barrier.lua:226]: Same for all the other lines

Comment (Toby1710) [reply mods/ctf/ctf_map/ctf_map_core/barrier.lua:226]: Why?

Comment (LoneWolfHT) [reply mods/ctf/ctf_map/ctf_map_core/barrier.lua:226]: That's how our code style works

Comment (Toby1710) [reply mods/ctf/ctf_map/ctf_map_core/barrier.lua:226]: Can you show how it should be? Maybe suggest changes?

Comment (LoneWolfHT) [inline mods/ctf/ctf_map/ctf_map_core/barrier.lua:226]:
```suggestion
			minetest.chat_send_player(name,
					minetest.colorize("#e32d2d", "Can't dig beyond the barrier!"))
```

File: mods/ctf/ctf_map/ctf_map_core/barrier.lua [hunk 2]
```diff
@@ -222,7 +222,8 @@ if minetest.get_modpath("ctf") then
 		local tname = ctf.player(name).team
 		if tname and
 				(tname == "blue" and pos.z >= 0) or (tname == "red" and pos.z <= 0) then
-			minetest.chat_send_player(name, "Can't dig beyond the barrier!")
+			minetest.chat_send_player(name,
+					minetest.colorize("#e32d2d", "Can't dig beyond the barrier!"))
```
Comment (TSafa-23) [inline mods/ctf/ctf_map/ctf_map_core/barrier.lua:226]:
```suggestion
			minetest.chat_send_player(name, minetest.colorize("#e32d2d", "Can't dig beyond the barrier!"))
```preferable imo

File: mods/ctf/ctf_match/buildtime.lua [hunk 3]
```diff
@@ -65,7 +65,8 @@ end)
 minetest.register_on_punchplayer(function(_, hitter)
 	if ctf_match.is_in_build_time() then
 		if hitter:is_player() then
-			minetest.chat_send_player(hitter:get_player_name(), "Match hasn't started yet!")
+			minetest.chat_send_player(hitter:get_player_name(),
+                    minetest.colorize("#e32d2d", "Match hasn't started yet!"))
```
Comment (TSafa-23) [inline mods/ctf/ctf_match/buildtime.lua:69]:
```suggestion
			minetest.chat_send_player(hitter:get_player_name(), minetest.colorize("#e32d2d", "Match hasn't started yet!"))
```

File: mods/ctf/ctf_match/buildtime.lua [hunk 4]
```diff
@@ -88,7 +89,8 @@ end)
 
 ctf_flag.register_on_prepick_up(function(name, flag)
 	if ctf_match.is_in_build_time() then
-		minetest.chat_send_player(name, "Match hasn't started yet!")
+		minetest.chat_send_player(name,
+                minetest.colorize("#e32d2d", "Match hasn't started yet!"))
```
Comment (TSafa-23) [inline mods/ctf/ctf_match/buildtime.lua:93]:
```suggestion
		minetest.chat_send_player(name, minetest.colorize("#e32d2d", "Match hasn't started yet!"))
```

**Review by TSafa-23 — CHANGES_REQUESTED** [review]
Green and yellow is not a very good combination, low contrast and do not go together. May I suggest something easier to read and to distinguish? Possibly red

**Comment by Toby1710** [comment]
> Green and yellow is not a very good combination, low contrast and do not go together. May I suggest something easier to read and to distinguish? Possibly red

What do you mean?

**Comment by xenonca** [comment]
This should better be displayed as a HUD message, not in chat.

**Comment by DiamondPlane** [comment]
> This should better be displayed as a HUD message, not in chat.

It already was in chat so I just colorized it.

**Comment by xenonca** [comment]
> > This should better be displayed as a HUD message, not in chat.
>
> It already was in chat so I just colorized it.

Added `hud_events` in the meanwhile so you might want to change that now.
Also I think this shouldn't be colorized red. In Minetest a red message is something you usually get when an error ocured, this is not the case here. Changing the color or even leaving it white is fine if it's added as a hud_event IMO

**Comment by DiamondPlane** [comment]
Ok, that's good that you added "hud_events". 😃
Maybe I'll not colorize it but make it as a hud?

**Pull request closed without merging**